src/services/formatting/formatting.ts

```diff
@@ -1083,14 +1098,19 @@ namespace ts.formatting {
             }
 
             // shift all parts on the delta size
-            const delta = indentation - nonWhitespaceColumnInFirstPart.column;
+            let delta = indentation - nonWhitespaceColumnInFirstPart.column;
             for (let i = startIndex; i < parts.length; i++ , startLine++) {
                 const startLinePos = getStartPositionOfLine(startLine, sourceFile);
                 const nonWhitespaceCharacterAndColumn =
                     i === 0
                         ? nonWhitespaceColumnInFirstPart
                         : SmartIndenter.findFirstNonWhitespaceCharacterAndColumn(parts[i].pos, parts[i].end, sourceFile, options);
-
+                if (jsxTextStyleIndent) {
+                    // skip adding indentation to blank lines

```

Comment:
So for comments we actually add trailing whitespace, apparently. This seems silly, and causes some jsx text formatting baselines to fail, so we choose not to emit any whitespace on empty lines here to avoid emitting that trailing whitespace.